backend/alert/views.py

```diff
@@ -507,7 +512,7 @@ def update(self, request, pk=None):
         return Response({"detail": "no changes made"}, status=status.HTTP_200_OK)
 
     def create(self, request, *args, **kwargs):
-        if self.get_queryset().filter(id=request.data.get("id")).exists():
+        if Registration.objects.filter(id=request.data.get("id")).exists():
             return self.update(request, request.data.get("id"))
```

Comment:
Consider if we find that an object has `id = request.data.get("id")` but that object is not `self`. `self` will be updated to have an `id` of `request.data.get("id")`, but what will the object that used to have an `id` of `request.data.get('id')` have?

Reply:
There's actually a check in the [update method](https://github.com/pennlabs/penn-courses/pull/309/files#diff-30321778847dd168745458b161cae6aa7914fecdc746484bb7267d2da2bdafd7R368) to throw a 404 (not found) response if an object with the given id doesn't exist in all `Regstration.objects`, and otherwise a 403 (forbidden) response if an object with the given id doesn't exist in the set of registrations that the user owns (the `queryset`), meaning a registration with the specified id exists but it isn't owned by the user who is requesting to modify it. 

This redirection to `self.update(request, request.data.get("id"))` if an object with the given id exists in all `Registration.objects` will cause an update if the user owns the registration (as the docs say it should), but if the user doesn't own the registration with the matching ID, it will return a 403. So there's no case where a registration id will actually be changed to an id that is already taken by a different registration (in which Django would throw an `IntegrityError` saying "UNIQUE constraint failed".